File: backend/app/Models/MenuItem.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class MenuItem extends Model
{
    use HasFactory;

    protected $fillable = [
      'name',
      'long_description',
      'short_description',
      'recipe',
      'picture',
      'price',
      'menu_id'
    ];

    public function menu()
    {
        return $this->belongsTo(Menu::class);
    }

    public function users()
    {
        return $this->belongsToMany(User::class)
            ->withPivot('quantity', 'price', 'menu_item_id')
            ->withTimestamps();
    }

    public function orders()
    {
        return $this->belongsToMany(Order::class, 'menu_items_orders')
            ->withPivot('quantity', 'price', 'order_id')
            ->withTimestamps();
    }

    public function favoritedByUsers()
    {
        return $this->belongsToMany(User::class, 'user_favorite_items');
    }

}

File: mobile/app/src/main/java/com/yehorsk/platea/menu/domain/models/MenuItem.kt
package com.yehorsk.platea.menu.domain.models

data class MenuItem(
    val createdAt: String,
    val updatedAt: String,
    val id: Int,
    val menuId: String,
    val name: String,
    val shortDescription: String,
    val longDescription: String,
    val recipe: String,
    val picture: String,
    val price: Double,
    val availability: Boolean,
    val isFavorite: Boolean = false
)
